src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -38,24 +39,62 @@ public SchemaProvider(MySqlConnection connection)
 			{ "Triggers", FillTriggers },
 			{ "UserPrivileges", FillUserPrivileges },
 			{ "Views", FillViews },
+			{ "ForeignKeys", FillForeignKeys },
+			{ "Indexes", FillIndexes },
+			{ "IndexColumns", FillIndexes },
 		};
 	}
 
-	public ValueTask<DataTable> GetSchemaAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) => GetSchemaAsync(ioBehavior, "MetaDataCollections", cancellationToken);
+	private async Task FillForeignKeys(IOBehavior ioBehavior, DataTable dataTable, string?[] restrictionValues, CancellationToken cancellationToken)
+	{
+		dataTable.Columns.AddRange(new DataColumn[]
+		{
+			new("CONSTRAINT_CATALOG", typeof(string)),
+			new("CONSTRAINT_SCHEMA", typeof(string)),
+			new("CONSTRAINT_NAME", typeof(string)),
+			new("TABLE_CATALOG", typeof(string)),
+			new("TABLE_SCHEMA", typeof(string)),
+			new("TABLE_NAME", typeof(string)),
+			new("COLUMN_NAME", typeof(string)),
+		});
+
+		await FillDataTableAsync(ioBehavior, dataTable, "KEY_COLUMN_USAGE", restrictionValues, cancellationToken).ConfigureAwait(false);
+	}
+
+	private async Task FillIndexes(IOBehavior ioBehavior, DataTable dataTable, string?[] restrictionValues, CancellationToken cancellationToken)
+	{
+		dataTable.Columns.AddRange(new DataColumn[]
+		{
+			new("TABLE_CATALOG", typeof(string)),
+			new("TABLE_SCHEMA", typeof(string)),
+			new("TABLE_NAME", typeof(string)),
+			new("NON_UNIQUE", typeof(string)),
+			new("INDEX_SCHEMA", typeof(string)),
+			new("INDEX_NAME", typeof(string)),
+			new("COLUMN_NAME", typeof(string)),
+			new("COLLATION", typeof(string)),
+			new("CARDINALITY", typeof(string)),
+		});
 
-	public async ValueTask<DataTable> GetSchemaAsync(IOBehavior ioBehavior, string collectionName, CancellationToken cancellationToken)
+		await FillDataTableAsync(ioBehavior, dataTable, "STATISTICS", restrictionValues, cancellationToken).ConfigureAwait(false);
+	}
+
+	public ValueTask<DataTable> GetSchemaAsync(IOBehavior ioBehavior, string?[] restrictionValues, CancellationToken cancellationToken) => GetSchemaAsync(ioBehavior, "MetaDataCollections", restrictionValues, cancellationToken);
+
+	public async ValueTask<DataTable> GetSchemaAsync(IOBehavior ioBehavior, string collectionName, string?[] restrictionValues, CancellationToken cancellationToken)
 	{
 		if (collectionName is null)
 			throw new ArgumentNullException(nameof(collectionName));
+		collectionName = collectionName.Replace(" ", "");
```

Comment:
What motivates this?

Reply:
`Foreign Keys or ForeignKeys`: NHibernate call this Methode with "Foreign Keys". I'm not sure which is the correct magic string variant.